### ISSUE TITLE
python: multi-line non-docstring string lines counted as blank in LOC

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#415]
## Summary

For Python, the interior lines of a multi-line string literal that is **not** a
bare docstring (i.e. assigned to a variable or passed as an argument) are
mis-counted as blank lines and excluded from ploc.

## Location

- `src/metrics/loc.rs:660-673` — `impl Loc for PythonCode` (the `String` arm and
  the `StringStart | StringEnd | StringContent` no-op arm at `loc.rs:661`)

## Evidence

The `String` arm inserts at most the string's **first** row into `ploc.lines`,
and only when `parent.start_row() != start`; the `StringStart` / `StringContent`
/ `StringEnd` leaf tokens are no-ops (`loc.rs:661`). So every interior physical
row of a triple-quoted string whose parent is not an `ExpressionStatement` is
never added to ploc, is not a comment, and therefore falls into
`blank = sloc − ploc − comment-only-lines`.

```python
QUERY = """
SELECT id, name
FROM users
WHERE active = 1
ORDER BY name
"""
```

Observed `sloc=6, ploc=1, lloc=1, cloc=0, blank=5`; expected `blank=0, ploc=6`.

## Expected Behavior

Physical lines occupied by a multi-line string literal are code lines (ploc),
not blank lines.

## Actual Behavior

5 of the 6 lines are reported as blank; ploc is 1.

## Impact

ploc and blank are wrong for the very common pattern of multi-line
SQL/templates/embedded text bound to a variable or passed as an argument. blank
is inflated and ploc deflated, skewing any ploc-derived reporting and averages.

## Related (observed, lower priority)

Docstring classification is inconsistent across metrics: a bare-string docstring
is counted as `cloc` (`loc.rs:667`) **and** increments `lloc` (the wrapping
`ExpressionStatement`, `loc.rs:695`), while Tokens counts a docstring as +3
tokens even though LOC treats that line as a comment. These are pre-existing,
partly-documented choices (`python_cloc` test) — noting for awareness; the
blank/ploc bug above is the actionable defect.

---

## Resolution

Fixed on branch `fix/issue-415` (commit bd624d8). The Python `String`
arm in `src/metrics/loc.rs` now inserts every interior row
`start + 1..=end` of a non-docstring multi-line string into
`ploc.lines`; the docstring branch is unchanged. The example
`QUERY = """…"""` now reports `ploc=6`, `blank=0`. See the comment
above for full details.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

`impl Loc for PythonCode` (`src/metrics/loc.rs:660-673`) inserts only the first
row of a `String` node into `ploc.lines` (and only when the parent starts on a
different row); the `StringStart`/`StringContent`/`StringEnd` leaves are no-ops.
So interior rows of a multi-line string are never marked as code and leak into
`blank`.

### Implementation Steps

1. In the `String` arm, for a non-docstring multi-line string (parent is not an
   `ExpressionStatement`), insert **every** row `start..=end` into
   `ploc.lines`, not just `start` — matching how multi-line constructs are
   handled elsewhere. (Alternatively, give `StringContent` a real arm that marks
   its rows as ploc.)
2. Keep the docstring branch (`parent == ExpressionStatement → add_cloc_lines`)
   unchanged unless the related docstring-classification issue is taken up
   separately.

### Tests to Add/Update

- A 6-line `QUERY = """…"""` → `ploc == 6`, `blank == 0`.
- A multi-line string passed as a call argument → its rows count as ploc.
- Existing `python_string_on_new_line` and docstring tests still pass.
- Anchor with `assert_eq!` on the loc accessors.

### Verification

- [ ] `cargo test -p big-code-analysis` passes
- [ ] `bca metrics` on the SQL repro returns `ploc=6, blank=0`
- [ ] Integration snapshots under the `big-code-analysis-output` submodule
      re-accepted if any Python fixture contains multi-line strings

**Comment by dekobon** [comment]
## Fixed in `fix/issue-415` (commit bd624d8)

**Root cause**
In `impl Loc for PythonCode` (`src/metrics/loc.rs`), the `String` arm
inserted at most the string's *first* row into `ploc.lines`, and only
when `parent.start_row() != start`. The `StringStart | StringEnd |
StringContent` tokens are no-ops. So every interior physical row of a
triple-quoted, non-docstring string was never added to `ploc`, was not
a comment, and fell into `blank = sloc - ploc - only_comment_lines`.

**Changes**
The non-docstring branch now inserts every interior/trailing row
`start + 1..=end` into `ploc.lines`, mirroring the existing
`PreprocArg` unexpanded-macro handling in the C `_` arm. The opening
row is still inserted only when the parent statement begins on an
earlier row (otherwise that row is already attributed to the enclosing
statement). The docstring branch (`parent == ExpressionStatement ->
add_cloc_lines`) is unchanged.

**Languages**
Python only. Other languages treat `String` as a structural no-op and
rely on a different child-handling path; they do not share the Python
docstring/non-docstring split. Out of scope for this issue.

**Tests** (all anchored with `assert_eq!` on loc accessors)
- `python_multiline_string_assignment_ploc`: the 6-line `QUERY = """…"""`
  case -> `ploc == 6`, `blank == 0`.
- `python_multiline_string_argument_ploc`: a multi-line string passed
  as a call argument -> all rows counted, `ploc == 4`, `blank == 0`.
- `python_single_line_string_assignment_ploc`: single-line string
  unchanged (`start == end`).
- `python_multiline_docstring_still_cloc`: a function docstring still
  counts as `cloc`, not blank.

Verified test-via-revert: the two ploc/blank tests fail against
pre-fix code (`ploc` 1 vs 6, 2 vs 4); the docstring test passes under
both, as expected.

**Snapshots affected**
In-tree unit snapshot for the new assignment test (anchored). No
integration snapshots in the `big-code-analysis-output` submodule
produced `.snap.new` — full `cargo test -p big-code-analysis
--all-features` is clean from a fresh tree.

The related docstring-classification inconsistency across metrics is
intentionally left out of scope per the issue.
